packages/myst-to-tex/src/index.ts

```diff
@@ -247,20 +248,30 @@ const handlers: Record<string, Handler> = {
   include(node, state) {
     state.renderChildren(node, true);
   },
+  footnoteReference(node, state) {
+    if (!node.identifier) return;
+    const footnote = state.references.footnotes?.[node.identifier];
```

Comment:
Here, we assume `footnoteDefinition` nodes have been pulled off the tree and are passed in separately. This is how `myst-cli` currently operates but is not super nice. We should move towards just leaving the footnote nodes in the tree rather than passing around a separate `references` object.